auth/api/iam/openid4vp.go

```diff
@@ -158,6 +163,133 @@ func (r Wrapper) handleAuthorizeRequestFromHolder(ctx context.Context, verifier
+	}
+	// check the response URL because later errors will redirect to this URL
+	responseURI, responseOK := params[responseURIParam]
+	if !responseOK {
+		return nil, oauthError(oauth.InvalidRequest, "missing response_uri parameter")
+	}
+	clientIDScheme, ok := params[clientIDSchemeParam]
+	if !ok || clientIDScheme != didScheme {
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.InvalidRequest, "invalid client_id_scheme parameter"), responseURI)
+	}
+	verifierID, ok := params[clientIDParam]
+	if !ok {
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.InvalidRequest, "missing client_id parameter"), responseURI)
+	}
+	// the verifier must be a did:web
+	verifierDID, err := did.ParseDID(verifierID)
+	if err != nil || verifierDID.Method != "web" {
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.InvalidRequest, "invalid client_id parameter (only did:web is supported)"), responseURI)
+	}
+	nonce, ok := params[nonceParam]
+	if !ok {
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.InvalidRequest, "missing nonce parameter"), responseURI)
+	}
+	// get verifier metadata
+	clientMetadataURI, ok := params[clientMetadataURIParam]
+	if !ok {
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.InvalidRequest, "missing client_metadata_uri parameter"), responseURI)
+	}
+	// we ignore any client_metadata, but officially an error must be returned when that param is present.
+	metadata, err := r.auth.Holder().ClientMetadata(ctx, clientMetadataURI)
+	if err != nil {
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.ServerError, "failed to get client metadata (verifier)"), responseURI)
+	}
+	// get presentation_definition from presentation_definition_uri
+	presentationDefinitionURI, ok := params[presentationDefUriParam]
+	if !ok {
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.InvalidRequest, "missing presentation_definition_uri parameter"), responseURI)
+	}
+	presentationDefinition, err := r.auth.Holder().PresentationDefinition(ctx, presentationDefinitionURI)
+	if err != nil {
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.InvalidPresentationDefinitionURI, fmt.Sprintf("failed to retrieve presentation definition on %s", presentationDefinitionURI)), responseURI)
+	}
+
+	// at this point in the flow it would be possible to ask the user to confirm the credentials to use
+
+	// all params checked, delegate responsibility to the holder
+	vp, submission, err := r.auth.Holder().BuildPresentation(ctx, walletDID, *presentationDefinition, metadata.VPFormats, nonce)
+	if err != nil {
+		if errors.Is(err, oauth2.ErrNoCredentials) {
+			return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.InvalidRequest, "no credentials available"), responseURI)
+		}
+		return r.sendAndHandleDirectPostError(ctx, oauthError(oauth.ServerError, err.Error()), responseURI)
+	}
+
+	// any error here is a server error, might need a fixup to prevent exposing to a user
```

Comment:
The how, depends. When I think of iDEAL; Probably a webpage that informs the user, with a link to return back to the verifier. If it was a pop-up; close window.

Reply:
This is true for other situations as well. We'll encounter them when creating the demo's